src/components/nps/nps.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { range } from 'lodash-es'
+import clsx from 'clsx'
+
+/**
+ * NPS widget properties
+ */
+export type NPSProps = {
+  /** Title message to show above NPS scale */
+  message?: string
+  /** Callback function which is triggered when a user selects score */
+  onSubmit?: (value: number) => void
+  /** Additional classes to apply on NPSInput component */
+  className?: string
+  /** Additional styles to apply on NPSInput component */
+  style?: React.CSSProperties
+}
+
+/**
+ * The Net Promoter Score (NPS) methodology is based on asking customers a single question
+ * that predicts the likelihood of both repurchase and referral: “How likely is it that you would recommend this company to a friend or colleague?” Customers rate their answers on a scale from 0 to 10.
+ *
+ * The answers customers provide are classified as follows:
+ *
+ *  * 0–6 = Detractors—unhappy customers who can hurt your brand through negative word-of-mouth
+ *  * 7–8 = Passives—satisfied but indifferent customers who could be swayed by the competition
+ *  * 9–10 = Promoters—loyal customers who will keep buying and referring others
+ *
+ * Finally, NPS score can be easily calculated using a mathematical formula
+ * ** ((Promoters - Detractors) / Respondents) * 100 **
+ */
+export function NPSInput({
+  message = 'How likely are you to recommend us to your friends and colleagues?',
+  onSubmit,
+  className,
+  style,
+}: NPSProps) {
+  const [npsScore, setNpsScore] = useState<number | null>(null)
```

Comment:
The component can be only used a controlled component, which is fine, but in this ui-kit we have almost every component which is behaving as a controlled or uncontrolled component. I would suggest to use something like

```jsx
const [score, setScore] = useSyncedState<number | undefined>(defaultValue || value || undefined)

// tempScore state is used to have the temporary score while hovering, where as score
// is set when the user selects a particular score by clicking
const [tempScore, setTempScore] = useState<number | undefined>(undefined)
```